GCE/sources/GCEResourceLimits.py

```diff
@@ -26,7 +26,9 @@ def acquire(self):
         """
 
         factory_data = super().acquire()
-        df_factory_data = factory_data.get(self.data_keys[0])
+        if len(factory_data) != 1:
+            raise RuntimeError("Incorrect number of elements in data block.")
```

Comment:
Should there be a test to cover this exception?

Reply:
Yes, I suppose so.  I'll see what I can come up with.

Reply:
I do not understand how this PR is related to framework issue 386 at all?  How could that change break a google self-test?

Reply:
@StevenCTimm, you're right that this check is not directly germane to the PR.  However, because the type of `factory_data` changed from a list to a dictionary, the way one accesses the data has changed (I opted for the `factory_data.popitem()` interface).  Using this functionality makes sense if there is only one item in the dictionary--hence the check.

Before this change the code only cared about the first element in the `self.data_keys` attribute.  Now the code *expects* only one element.

Reply:
Thinking about this further, the check I was making was actually a configuration check, and I've moved the check to the module's constructor.